pandas/tests/generic/methods/test_reorder_levels.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pytest
+
+from pandas import DataFrame, MultiIndex, Series
+import pandas._testing as tm
+
+
+class TestReorderLevels:
+    @pytest.mark.parametrize("klass", [Series, DataFrame])
+    def test_reorder_levels(self, klass):
+        index = MultiIndex(
+            levels=[["bar"], ["one", "two", "three"], [0, 1]],
+            codes=[[0, 0, 0, 0, 0, 0], [0, 1, 2, 0, 1, 2], [0, 1, 0, 1, 0, 1]],
+            names=["L0", "L1", "L2"],
+        )
+        df = DataFrame({"A": np.arange(6), "B": np.arange(6)}, index=index)
+        obj = df if klass is DataFrame else df["A"]
+
+        # no change, position
+        result = obj.reorder_levels([0, 1, 2])
+        tm.assert_equal(obj, result)
+
+        # no change, labels
+        result = obj.reorder_levels(["L0", "L1", "L2"])
+        tm.assert_equal(obj, result)
+
+        # rotate, position
+        result = obj.reorder_levels([1, 2, 0])
+        e_idx = MultiIndex(
+            levels=[["one", "two", "three"], [0, 1], ["bar"]],
+            codes=[[0, 1, 2, 0, 1, 2], [0, 1, 0, 1, 0, 1], [0, 0, 0, 0, 0, 0]],
+            names=["L1", "L2", "L0"],
+        )
+        expected = DataFrame({"A": np.arange(6), "B": np.arange(6)}, index=e_idx)
+        expected = expected if klass is DataFrame else expected["A"]
+        tm.assert_equal(result, expected)
+
+        result = obj.reorder_levels([0, 0, 0])
+        e_idx = MultiIndex(
+            levels=[["bar"], ["bar"], ["bar"]],
+            codes=[[0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0]],
+            names=["L0", "L0", "L0"],
+        )
+        expected = DataFrame({"A": np.arange(6), "B": np.arange(6)}, index=e_idx)
+        expected = expected if klass is DataFrame else expected["A"]
+        tm.assert_equal(result, expected)
+
+        result = obj.reorder_levels(["L0", "L0", "L0"])
+        tm.assert_equal(result, expected)
+
+    def test_reorder_levels_swaplevel_equivalence(self):
+
+        ymd = tm.makeTimeDataFrame(100)
+        mi = MultiIndex.from_arrays(
+            [ymd.index.year, ymd.index.month, ymd.index.day],
+            names=["year", "month", "day"],
+        )
+        ymd.index = mi
```

Comment:
is this the same as multiindex_year_month_day_dataframe_random_data fixture?

Reply:
yes, will update